vividus-facade-jira/src/main/java/org/vividus/jira/JiraFacade.java

```diff
@@ -38,49 +40,55 @@
     private static final String ISSUE = "issue/";
     private static final String ISSUE_ENDPOINT = REST_API_ENDPOINT + ISSUE;
 
-    private final JiraClient jiraClient;
+    private final JiraClientProvider jiraClientProvider;
 
-    public JiraFacade(JiraClient jiraClient)
+    public JiraFacade(JiraClientProvider jiraClientProvider)
     {
-        this.jiraClient = jiraClient;
+        this.jiraClientProvider = jiraClientProvider;
     }
 
-    public String createIssue(String issueBody) throws IOException
+    public String createIssue(String issueBody, Optional<String> jiraKey) throws IOException, JiraConfigurationException
     {
-        return jiraClient.executePost(ISSUE_ENDPOINT, issueBody);
+        return jiraClientProvider.get(null, jiraKey).executePost(ISSUE_ENDPOINT, issueBody);
     }
 
-    public String updateIssue(String issueKey, String issueBody) throws IOException
+    public String updateIssue(String issueKey, String issueBody, Optional<String> jiraKey)
```

Comment:
what's the real usecase here? when both `issueKey` and `jiraKey` are needed?

Reply:
Previously I proposed to use Either, but we may want to dynamically evaluate jira instance based on jira key or to statically couple all keys to one jira instance to avoid incertainty

Reply:
will be removed for updates